opm/parser/eclipse/Parser/Parser.cpp

```diff
@@ -42,60 +42,194 @@
 #include <opm/parser/eclipse/RawDeck/RawConsts.hpp>
 #include <opm/parser/eclipse/RawDeck/RawEnums.hpp>
 #include <opm/parser/eclipse/RawDeck/RawKeyword.hpp>
+#include <opm/parser/eclipse/Utility/Stringview.hpp>
 
 namespace Opm {
 
+    template< typename Itr >
+    static inline Itr find_comment( Itr begin, Itr end ) {
+
+        auto itr = std::find( begin, end, '-' );
+        for( ; itr != end; itr = std::find( itr + 1, end, '-' ) )
+            if( (itr + 1) != end &&  *( itr + 1 ) == '-' ) return itr;
+
+        return end;
+    }
+
+    template< typename Itr, typename Term >
+    static inline Itr strip_after( Itr begin, Itr end, Term terminator ) {
```

Comment:
Not really. It looks for some terminator, either terminating slash *or* a comment (it is the shared implementation for both), but respects that both constructs can be present in quotes. It also properly returns the interesting range. It's really a `takeWhile` that respects quotes.